packages/hawtio/src/preferences/HomePreferences.tsx

```diff
@@ -1,13 +1,46 @@
 import { Button, CardBody, Form, FormGroup, FormSection, Switch } from '@patternfly/react-core'
 import React, { useState } from 'react'
-import { log } from './globals'
+
+type LocalStorageFieldData<T> = {
```

Comment:
It was just to comply with TS types because if I wanted to do a decorator for savingToLocalStorage function I do need to save the type if I don't want to just use any and remove all type checking.

I mean, there is a much easier way to do this just by doing something like

(value) => {saveToStorage(key, value); updateValue(value)} in the component but I was thinking in generalizing that. (Although I don't know if that would make react compain... any way, I'll be rewriting this using a service as you said below so it will be much simpler)